### ISSUE TITLE
spring-boot 2.2.1 and bowman 0.8.0 -> ClassNotFoundException 

### ISSUE THREAD

**Issue by dkrizic** [BlackPepperSoftware/bowman#38]
I have a client app that is based on spring-boot-starter-parent:2.2.1.RELEASE and I have added bowman:0.8.0 to this application. At runtime I get this exception:

```
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'createClientFactory' defined in class path resource [com/prodyna/pac/conference/frontend/client/BackendClientFactory.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [uk.co.blackpepper.bowman.ClientFactory]: Factory method 'createClientFactory' threw exception; nested exception is java.lang.NoClassDefFoundError: org/springframework/hateoas/hal/Jackson2HalModule
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:645) ~[spring-beans-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:475) ~[spring-beans-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1338) ~[spring-beans-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1177) ~[spring-beans-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:557) ~[spring-beans-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:517) ~[spring-beans-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:323) ~[spring-beans-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:222) ~[spring-beans-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:321) ~[spring-beans-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202) ~[spring-beans-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276) ~[spring-beans-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1287) ~[spring-beans-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1207) ~[spring-beans-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:636) ~[spring-beans-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	... 63 common frames omitted
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [uk.co.blackpepper.bowman.ClientFactory]: Factory method 'createClientFactory' threw exception; nested exception is java.lang.NoClassDefFoundError: org/springframework/hateoas/hal/Jackson2HalModule
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:185) ~[spring-beans-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:640) ~[spring-beans-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	... 76 common frames omitted
Caused by: java.lang.NoClassDefFoundError: org/springframework/hateoas/hal/Jackson2HalModule
	at uk.co.blackpepper.bowman.RestOperationsFactory.<init>(RestOperationsFactory.java:117) ~[bowman-client-0.8.0.jar:na]
	at uk.co.blackpepper.bowman.ClientFactory.<init>(ClientFactory.java:42) ~[bowman-client-0.8.0.jar:na]
	at uk.co.blackpepper.bowman.ClientFactory.<init>(ClientFactory.java:35) ~[bowman-client-0.8.0.jar:na]
	at uk.co.blackpepper.bowman.Configuration.buildClientFactory(Configuration.java:166) ~[bowman-client-0.8.0.jar:na]
	at com.prodyna.pac.conference.frontend.client.BackendClientFactory.createClientFactory(BackendClientFactory.java:25) ~[classes/:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:567) ~[na:na]
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:154) ~[spring-beans-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	... 77 common frames omitted
Caused by: java.lang.ClassNotFoundException: org.springframework.hateoas.hal.Jackson2HalModule
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:602) ~[na:na]
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178) ~[na:na]
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521) ~[na:na]
	... 87 common frames omitted
```

The client factory does nothing special

```
    @Bean
    public ClientFactory createClientFactory() {
        return Configuration.builder().setBaseUri(backendAddress.getURI()).build()
                .buildClientFactory();
    }
```

I compile using Java 13. Is this a known problem. Spring Boot 2.1.3 with bowman 0.3.0 was working properly

**Comment by dkrizic** [comment]
bowman 0.8.1-SNAPSHOT depends on spring-boot-depencencies 2.1.6.RELEASE resulting in spring-hateoas 1.0.1.RELEASE. But spring-hateoas in this version has no package called org.springframework.hateoas.hal (see https://github.com/spring-projects/spring-hateoas/tree/1.0.1.RELEASE/src/main/java/org/springframework/hateoas). Older versions (0.2.x) did however (see https://github.com/spring-projects/spring-hateoas/tree/0.25.1.RELEASE/src/main/java/org/springframework/hateoas/hal). How is this supposed to work at all?

**Comment by dkrizic** [comment]
Ah, got it.

Spring-Boot 2.1.x -> Hateoas 0.25.x (with HAL)
Spring-Boot 2.2.x -> Hateoas 1.0.1 (without HAL)

Can you please provide a new version that is compatible with Spring Boot 2.2.x?

**Comment by hdpe** [comment]
Fixed in #37 

Thanks

**Comment by hdpe** [comment]
Released in `0.9.0`.

**Comment by dkrizic** [comment]
Thank you!
